plugins/kubernetes/app/controllers/kubernetes_dashboard_controller.rb

```diff
@@ -0,0 +1,82 @@
+require 'kubeclient'
+
+class KubernetesDashboardController < ApplicationController
+  include CurrentProject
+
+  def index
+    render json: format_pod_info(pod_info).to_json
+  end
+
+  private
+
+  def pod_info
+    roles = {}
+    Environment.find(params[:environment]).cluster_deploy_groups.each do |cluster_deploy_group|
+      cluster_deploy_group.cluster.client.get_pods(
+          namespace: cluster_deploy_group.namespace,
+          label_selector: "project=#{current_project.name_for_label}").each do |pod|
+        labels = pod.metadata.labels
+        roles[labels.role_id] ||= build_role(labels.role_id)
+        role = roles[labels.role_id]
```

Comment:
Could we move this two lines into the `build_role` method and have the`build_role` method setting the role in the hash and returning it? It would simplify this logic.